### ISSUE TITLE
FW: User ccrown uploaded usage file for 07/12/2020 Master for Site Towne Answering Service with missing metric headers/columns

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1900]
In GitLab by @kdjstudios on Jul 13, 2020, 11:49

**Submitted by:** harlie Crown <ccrown@towneanswering.com>

**Helpdesk:** https://www-servicedesk.answernet.com/profiles/ticket/2020-07-13-98323

**Server:** External

**Client/Site:** Towne

**Account:** NA

**Issue:** 

We are attempting to do billing today. I get this error message. The supposed missing header is in column CL right where it has always been.

The header is there, there is data in the column also.

Can you please fix this? I guess it’s related to the upgrade over the weekend…

![image](/uploads/29e0020edba7a1a9e4f83874810e89f0/image.png)

The uploaded file was missing the following headers:

·  PNBLSMsgInsAPI
Please ensure you uploaded the correct usage file. You may want to revert and re-upload or you may continue to process your billing without the usage for the above.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 13, 2020, 11:51

@amishra @rranjan @ranand @Robleski 

I was able to download the uploaded usage file via the Download button in SA Billing and can confirm that in fact it does have the CL column and the label matches exactly. So why is the client receiving this error?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 13, 2020, 11:54

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 13, 2020, 11:54

[billing_export_07_13_20.csv.20200713133028.20200713133028.csv](/uploads/891cb7ce05193af76449ab3808266c47/billing_export_07_13_20.csv.20200713133028.20200713133028.csv)

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jul 13, 2020, 12:14

@kdjstudios @Robleski 

Please check, it's been fixed.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 13, 2020, 12:18

I wrote:

> The support team found what was causing the error and they have updated it. May you please revert and re-upload your usage file to confirm it is working for you. Please let us know if all is working correctly. 


@ranand 

May someone please explain what caused this and how it was resolved. Thank you.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jul 14, 2020, 05:05

@kdjstudios @Robleski 

Extra space in the Towne's switch header element got introduced at the time of 11-July-2020 Release. We just removed that one extra space from the header element in the code.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on Jul 14, 2020, 09:10

so this was not found in testing by you?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 14, 2020, 11:45

@ranand @Robleski 

I think Greg's question is directed at the Aavaz team. But I can tell you that in the file I downloaded from SAB I did not "see" any extra spaces, but I also may not know how to look for those either.

EDIT: I see this was marked for approval, but I do not see what is needing to be approved? Was this not already released and updated on production, or please explain what else is needed to be performed.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jul 15, 2020, 04:49

@kdjstudios @Robleski 

* Missing header feature was not the part of 11-July-2020 Release. So, this was not tested on 11-July-2020
* Prior to 11-July-2020 Release it was working fine.
* Because it was one extra space, it got missed at the time of 11-July-2020 Release.

> But I can tell you that in the file I downloaded from SAB I did not "see" any extra spaces, but I also may not know how to look for those either.

That extra space was in the switch's required header element, which we check against the usage file headers.

> I see this was marked for approval, but I do not see what is needing to be approved? Was this not already released and updated on production, or please explain what else is needed to be performed.

`We removed that extra space from the production itself. We don't need to do any other development on this ticket. So, I marked it for PL-ForApproval to close this issue.`

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jul 15, 2020, 05:05

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on Jul 15, 2020, 05:54

kyle - do u need to test?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 15, 2020, 08:38

@Robleski 

So just to clarify, this was due to the Aavaz team putting an extra space in their code, not an extra space in the clients usage file. This would mean that this was not tested before it was released for this client. I know I only tested this functionality this on the Demo site. 

If I am understanding the Aavaz teams update, they have already released the fix on production so there is nothing to test.
